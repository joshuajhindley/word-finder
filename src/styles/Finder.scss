$light-color: #ddd;
$dark-color: #333;
$font-family: Roboto, sans-serif;

.root {
  height: 100vh;
  margin-top: 50px;
}

.modal-overlay {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.5);
  z-index: 1;
  cursor: pointer;

  .help-modal {
    position: absolute;
    top: 10%;
    left: 50%;
    transform: translate(-50%, -10%);
    width: 400px;
    max-width: 90vw;
    background-color: $light-color;
    color: $dark-color;
    border-radius: 5px;
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.5);
    z-index: 2;
    padding: 10px;
    cursor: default;

    &.dark {
      background-color: $dark-color;
      color: $light-color;
    }

    .help-modal-header {
      height: 50px;
      border-top-left-radius: 5px;
      border-top-right-radius: 5px;
      line-height: 14px;
      margin-top: 0;
      padding: 0 20px;
      display: flex;
      align-items: center;
      justify-content: space-between;

      .help-modal-header-title {
        font-size: 20px;
        font-weight: bold;
      }

      .help-modal-header-close {
        cursor: pointer;
        font-size: 20px;
      }
    }

    .help-modal-body {
      height: calc(100% - 50px);
      padding: 20px;
      overflow-y: auto;
      color: inherit;
      line-height: 20px;
      font-family: $font-family;
    }
  }
}

.app {
  background-color: $light-color;
  color: $dark-color;
  font-family: $font-family;
  min-height: 100vh;
  overflow: auto;
  transition: all 0.5s linear;

  button {
    background-color: $light-color;
    color: $dark-color;
    font-family: $font-family;
    transition: all 0.5s linear;
  }

  .header {
    height: 50px;
    align-content: center;
    justify-content: center;
    display: flex;
    flex: 1;

    a {
      margin-left: -10px;
      margin-top: 20px;

      button {
        cursor: pointer;
        border: 0;
        font-size: 16px;
        height: 30px;
        width: 70px;
      }
    }

    .help-message {
      position: absolute;
      margin-left: 150px;
      margin-top: 20px;
      font-size: 30px;
      width: 30px;
      height: 30px;
      background-color: $light-color;
      color: $dark-color;
      cursor: pointer;
      transition: all 0.5s linear;

      &.dark {
        background-color: $dark-color;
        color: $light-color;
        transition: all 0.5s linear;
      }
    }

    .color {
      box-shadow: inset 7px -4px 0 1px $dark-color;
      border-radius: 50%;
      cursor: pointer;
      width: 30px;
      height: 30px;
      margin-left: 175px;
      margin-top: 20px;
      transition: all 0.5s linear;

      &.dark {
        box-shadow: none;
        background-color: #fdd835;
        transition: all 0.5s linear;
      }
    }
  }

  .body {
    text-align: center;

    .search {
      cursor: default;

      h2 {
        margin-top: 20px;
      }

      h4 {
        margin: 20px auto 10px;
      }

      input {
        width: 45px;
        height: 45px;
        font-size: 26px;
        text-align: center;
        padding: 0;
        margin-left: 1px;
        margin-right: 1px;
        border-radius: 5px;
        border: 1px solid;
        background-color: white;

        &:focus {
          @media screen and (min-width: 768px) {
            outline: none;
          }
        }
      }

      button {
        border: 2px solid $dark-color;
        border-radius: 5px;
        cursor: pointer;
        font-size: 20px;
        margin-top: 40px;
        height: 50px;
        width: 200px;
      }

      .reset-button {
        margin-top: 10px;
        height: 40px;
        width: 150px;
        font-size: 18px;
      }
    }

    .results {
      margin-top: 40px;
      margin-bottom: 25px;

      .matches {
        margin-bottom: 10px;
      }

      .result {
        line-height: 22px;

        label {
          cursor: text;
        }
      }
    }
  }

  &.dark {
    background-color: $dark-color;
    color: $light-color;
    transition: all 0.5s linear;

    button {
      background-color: $dark-color;
      color: $light-color;
      transition: all 0.5s linear;
    }

    .body .search {
      button {
        border: 2px solid $light-color;
      }

      input {
        border-color: $dark-color;
      }
    }
  }
}

.scroll-to-top {
  opacity: 0;
  color: $dark-color;
  text-align: center;
  font-size: 50px;
  position: fixed;
  bottom: 50px;
  right: 50px;
  width: 10px;
  height: 10px;
  transition: opacity 0.25s linear;

  &.visible {
    opacity: 1;
    cursor: pointer;
    transition: opacity 0.25s linear;
  }

  &.dark {
    color: $light-color;
  }
}
