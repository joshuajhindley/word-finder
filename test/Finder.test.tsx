import React from 'react'
import Finder from '../src/ui/Finder'
import { mount } from 'enzyme'
import userEvent from '@testing-library/user-event'

describe('tests the FinderUI component', () => {
  Object.defineProperty(window, 'scrollTo', {
    writable: true,
    value: jest.fn().mockImplementation((values) => {
      global.scrollY = values.top
    }),
  })

  const props = {
    actions: {
      resetResults: jest.fn(),
      findResults: jest.fn(),
      invertColorMode: jest.fn(),
      toggleHelp: jest.fn(),
      toggleInPosition: jest.fn(),
    },
    results: [],
    isFresh: true,
    notInPositions: true,
    darkMode: true,
  }

  beforeEach(() => {
    props.actions.resetResults.mockClear()
    props.actions.findResults.mockClear()
    props.actions.invertColorMode.mockClear()
    props.actions.toggleHelp.mockClear()
    props.actions.toggleInPosition.mockClear()

    document.body.innerHTML = '<div id="root"></div>'
  })

  it('matches the snapshot', () => {
    const wrapper = mount(<Finder {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders in dark mode without crashing', () => {
    const wrapper = mount(<Finder {...props} />)
    expect(wrapper.find('.app').length).toBe(1)
    expect(wrapper.find('.header').length).toBe(1)
    expect(wrapper.find('.body').length).toBe(1)
    expect(wrapper.find('input').length).toBe(12)
    expect(wrapper.find('.results').length).toBe(1)
    expect(wrapper.find('.no-result').length).toBe(0)
    expect(wrapper.find('.result').length).toBe(0)
    expect(wrapper.find('.dark').length).toBe(3)
  })

  it('renders in light mode without crashing', () => {
    const wrapper = mount(<Finder {...props} darkMode={false} />)
    expect(wrapper.find('.app').length).toBe(1)
    expect(wrapper.find('.header').length).toBe(1)
    expect(wrapper.find('.body').length).toBe(1)
    expect(wrapper.find('input').length).toBe(12)
    expect(wrapper.find('.results').length).toBe(1)
    expect(wrapper.find('.no-result').length).toBe(0)
    expect(wrapper.find('.result').length).toBe(0)
    expect(wrapper.find('.dark').length).toBe(0)
  })

  it('calls the toggleHelp action when the help button is clicked', () => {
    const wrapper = mount(<Finder {...props} />)
    wrapper.find('.help-message').simulate('click')
    expect(props.actions.toggleHelp).toHaveBeenCalledTimes(1)
  })

  it('calls the toggleInPosition action when the checkbox is clicked', () => {
    const wrapper = mount(<Finder {...props} />)
    wrapper.find('.checkbox').simulate('click')
    expect(props.actions.toggleInPosition).toHaveBeenCalledTimes(1)
  })

  it('renders a no result message when no results are found', () => {
    const wrapper = mount(<Finder {...props} />)
    wrapper.find('.search-button').simulate('submit')
    expect(props.actions.findResults).toHaveBeenCalledTimes(1)
    expect(props.actions.findResults).toHaveBeenCalledWith({})
    wrapper.setProps({ results: [], isFresh: false })
    expect(wrapper.find('.no-result').length).toBe(1)
    expect(wrapper.find('.result').length).toBe(0)
  })

  it('renders a result message when results are found', () => {
    const wrapper = mount(<Finder {...props} />)
    wrapper.find('.color').simulate('click')
    expect(props.actions.invertColorMode).toHaveBeenCalledTimes(1)
  })

  it('renders a result message when results are found', () => {
    const wrapper = mount(<Finder {...props} />)
    wrapper.find('.search-button').simulate('submit')
    expect(props.actions.findResults).toHaveBeenCalledTimes(1)
    expect(props.actions.findResults).toHaveBeenCalledWith({})
    wrapper.setProps({
      results: ['WORDS', 'COINS', 'TESTS', 'VALUE', 'WORKS'],
      isFresh: false,
    })
    expect(wrapper.find('.no-result').length).toBe(0)
    expect(wrapper.find('.result').length).toBe(5)
    expect(wrapper.find('.matches').text()).toBe('Showing 5 matching words.')

    wrapper.find('.search-button').simulate('submit')
    expect(props.actions.findResults).toHaveBeenCalledTimes(2)
    expect(props.actions.findResults).toHaveBeenCalledWith({})
    wrapper.setProps({ results: ['WORDS'], isFresh: false })
    expect(wrapper.find('.no-result').length).toBe(0)
    expect(wrapper.find('.result').length).toBe(1)
    expect(wrapper.find('.matches').text()).toBe('Showing 1 matching word.')

    wrapper.find('.search-button').simulate('submit')
    expect(props.actions.findResults).toHaveBeenCalledTimes(3)
    expect(props.actions.findResults).toHaveBeenCalledWith({})
    wrapper.setProps({ results: [...Array(500)], isFresh: false })
    expect(wrapper.find('.no-result').length).toBe(0)
    expect(wrapper.find('.result').length).toBe(500)
    expect(wrapper.find('.matches').text()).toBe(
      'Showing first 500 matching words.'
    )
  })

  it('resets the results when the reset button is clicked', () => {
    const wrapper = mount(<Finder {...props} />)
    wrapper.find('.search-button').simulate('submit')
    expect(props.actions.findResults).toHaveBeenCalledTimes(1)
    expect(props.actions.findResults).toHaveBeenCalledWith({})
    wrapper.setProps({
      results: ['WORDS', 'COINS', 'TESTS', 'VALUE', 'WORKS'],
      isFresh: false,
    })
    expect(wrapper.find('.no-result').length).toBe(0)
    expect(wrapper.find('.result').length).toBe(5)
    wrapper.find('.reset-button').simulate('click')
    expect(props.actions.resetResults).toHaveBeenCalledTimes(1)
    wrapper.setProps({ results: [], isFresh: true })
    expect(wrapper.find('.no-result').length).toBe(0)
    expect(wrapper.find('.result').length).toBe(0)
  })

  it('scrolls to the input section of the page when an input is selected', () => {
    const wrapper = mount(<Finder {...props} />)
    const letter1 = wrapper.find('ForwardRef(Field)').first()
    expect(global.scrollY).toBe(0)
    letter1.simulate('focus')
    expect(global.scrollY).toBe(100)
  })

  it('navigates to different elements when navigation keys are pressed', () => {
    const wrapper = mount(<Finder {...props} />, {
      attachTo: document.getElementById('root'),
    })

    const letter1 = wrapper
      .find('ForwardRef(Field)')
      .at(0)
      .getDOMNode() as HTMLInputElement
    const letter2 = wrapper
      .find('ForwardRef(Field)')
      .at(1)
      .getDOMNode() as HTMLInputElement
    const letter3 = wrapper
      .find('ForwardRef(Field)')
      .at(2)
      .getDOMNode() as HTMLInputElement
    const letter4 = wrapper
      .find('ForwardRef(Field)')
      .at(3)
      .getDOMNode() as HTMLInputElement
    const letter5 = wrapper
      .find('ForwardRef(Field)')
      .at(4)
      .getDOMNode() as HTMLInputElement
    const input1 = wrapper.find('input').at(0)
    const input2 = wrapper.find('input').at(1)
    const input3 = wrapper.find('input').at(2)
    const input4 = wrapper.find('input').at(3)
    const input5 = wrapper.find('input').at(4)

    letter1.focus()
    expect(letter1).toHaveFocus()

    input1.simulate('keyup', { nativeEvent: { key: 'ArrowLeft' } })
    expect(letter1).toHaveFocus()
    input1.simulate('keyup', { nativeEvent: { key: 'ArrowRight' } })
    expect(letter2).toHaveFocus()
    input2.simulate('keyup', { nativeEvent: { key: 'ArrowRight' } })
    expect(letter3).toHaveFocus()
    input3.simulate('keyup', { nativeEvent: { key: 'ArrowRight' } })
    expect(letter4).toHaveFocus()
    input4.simulate('keyup', { nativeEvent: { key: 'ArrowRight' } })
    expect(letter5).toHaveFocus()
    input5.simulate('keyup', { nativeEvent: { key: 'ArrowRight' } })
    expect(letter5).toHaveFocus()
    input5.simulate('keyup', { nativeEvent: { key: 'ArrowLeft' } })
    input4.simulate('keyup', { nativeEvent: { key: 'Delete' } })
    input3.simulate('keyup', { nativeEvent: { key: 'Backspace' } })
    expect(letter2).toHaveFocus()
  })

  it('handles input correctly for the correct position inputs', (done) => {
    const wrapper = mount(<Finder {...props} />, {
      attachTo: document.getElementById('root'),
    })
    const letter1 = wrapper
      .find('ForwardRef(Field)')
      .at(0)
      .getDOMNode() as HTMLInputElement
    const letter2 = wrapper
      .find('ForwardRef(Field)')
      .at(1)
      .getDOMNode() as HTMLInputElement
    const letter5 = wrapper
      .find('ForwardRef(Field)')
      .at(4)
      .getDOMNode() as HTMLInputElement

    letter1.focus()

    expect(letter1).toHaveFocus()
    userEvent.keyboard('a')
    setTimeout(() => {
      wrapper.update()
      expect(wrapper.find('input').at(0).prop('value')).toBe('A')
      expect(letter2).toHaveFocus()

      userEvent.keyboard('[[')
      wrapper.update()
      expect(wrapper.find('input').at(1).prop('value')).toBe('')
      expect(letter2).toHaveFocus()
      userEvent.keyboard(' ')
      wrapper.update()
      expect(wrapper.find('input').at(1).prop('value')).toBe('')
      expect(letter2).toHaveFocus()
      userEvent.keyboard('1')
      wrapper.update()
      expect(wrapper.find('input').at(1).prop('value')).toBe('')
      expect(letter2).toHaveFocus()

      letter5.focus()
      userEvent.keyboard('z')
      setTimeout(() => {
        wrapper.update()
        expect(wrapper.find('input').at(4).prop('value')).toBe('Z')
        expect(letter5).toHaveFocus()
        done()
      }, 10)
    }, 0)
  })

  it('handles input correctly for the anyPos inputs', (done) => {
    const wrapper = mount(<Finder {...props} />, {
      attachTo: document.getElementById('root'),
    })

    const letter1 = wrapper
      .find('ForwardRef(Field)')
      .at(5)
      .getDOMNode() as HTMLInputElement
    const letter2 = wrapper
      .find('ForwardRef(Field)')
      .at(6)
      .getDOMNode() as HTMLInputElement
    const letter5 = wrapper
      .find('ForwardRef(Field)')
      .at(9)
      .getDOMNode() as HTMLInputElement
    const input1 = wrapper.find('input').at(5)
    const input2 = wrapper.find('input').at(6)

    letter1.focus()
    expect(letter1).toHaveFocus()

    input1.simulate('keyup', { nativeEvent: { key: 'ArrowLeft' } })
    expect(letter1).toHaveFocus()
    input1.simulate('keyup', { nativeEvent: { key: 'ArrowRight' } })
    expect(letter2).toHaveFocus()
    input2.simulate('keyup', { nativeEvent: { key: 'Delete' } })
    expect(letter1).toHaveFocus()

    userEvent.keyboard('a')
    setTimeout(() => {
      wrapper.update()
      expect(wrapper.find('input').at(5).prop('value')).toBe('A')
      expect(letter2).toHaveFocus()

      userEvent.keyboard(' ')
      wrapper.update()
      expect(wrapper.find('input').at(6).prop('value')).toBe('')
      expect(letter2).toHaveFocus()

      letter5.focus()
      userEvent.keyboard('z')
      setTimeout(() => {
        wrapper.update()
        expect(wrapper.find('input').at(9).prop('value')).toBe('Z')
        expect(letter5).toHaveFocus()
        done()
      }, 10)
    }, 0)
  })

  it('adds a new notLetter input when a letter is typed', (done) => {
    const wrapper = mount(<Finder {...props} />, {
      attachTo: document.getElementById('root'),
    })

    const letter1 = wrapper
      .find('ForwardRef(Field)')
      .at(11)
      .getDOMNode() as HTMLInputElement
    const input1 = wrapper.find('input').at(11)

    letter1.focus()
    expect(wrapper.find('input').length).toBe(12)
    expect(wrapper.find('ForwardRef(Field)').at(11).getDOMNode()).toHaveFocus()
    expect(letter1).toHaveFocus()
    expect(wrapper.find({ name: 'notLetter1' }).length).toBe(2)
    expect(wrapper.find({ name: 'notLetter2' }).length).toBe(0)
    input1.simulate('keyup', { nativeEvent: { key: 'ArrowLeft' } })
    expect(letter1).toHaveFocus()
    userEvent.keyboard('z')
    setTimeout(() => {
      wrapper.update()
      expect(wrapper.find('input').at(11).prop('value')).toBe('Z')
      expect(wrapper.find('input').length).toBe(13)
      expect(wrapper.find({ name: 'notLetter1' }).length).toBe(2)
      expect(wrapper.find({ name: 'notLetter2' }).length).toBe(2)

      const letter2 = wrapper
        .find('ForwardRef(Field)')
        .at(12)
        .getDOMNode() as HTMLInputElement
      const input2 = wrapper.find('input').at(12)

      expect(letter2).toHaveFocus()

      userEvent.keyboard('z')
      setTimeout(() => {
        expect(letter2).toHaveFocus()
        expect(wrapper.find('input').at(12).prop('value')).toBe('')
        expect(wrapper.find('input').length).toBe(13)

        input2.simulate('keyup', { nativeEvent: { key: 'ArrowLeft' } })
        expect(letter1).toHaveFocus()
        input1.simulate('keyup', { nativeEvent: { key: 'ArrowLeft' } })
        expect(letter1).toHaveFocus()
        input1.simulate('keyup', { nativeEvent: { key: 'ArrowRight' } })
        expect(letter2).toHaveFocus()
        input2.simulate('keyup', { nativeEvent: { key: 'Backspace' } })
        expect(letter1).toHaveFocus()
        input1.simulate('keyup', { nativeEvent: { key: 'Delete' } })
        expect(letter2).toHaveFocus()
        expect(wrapper.find('input').length).toBe(12)
        expect(wrapper.find({ name: 'notLetter1' }).length).toBe(0)
        expect(wrapper.find({ name: 'notLetter2' }).length).toBe(2)

        userEvent.keyboard(' ')
        wrapper.update()
        expect(wrapper.find('input').at(11).prop('value')).toBe('')
        expect(letter2).toHaveFocus()

        done()
      }, 10)
    }, 10)
  })
})
