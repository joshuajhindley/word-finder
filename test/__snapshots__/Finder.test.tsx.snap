// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the FinderUI component matches the snapshot 1`] = `
<Finder
  actions={
    Object {
      "findResults": [MockFunction],
      "invertColorMode": [MockFunction],
      "resetResults": [MockFunction],
      "toggleHelp": [MockFunction],
    }
  }
  darkMode={true}
  isFresh={true}
  results={Array []}
>
  <div
    className="app dark"
  >
    <div
      className="header"
    >
      <a
        href="/"
        tabIndex={-1}
      >
        <button
          tabIndex={-1}
        >
          &lt; Home
        </button>
      </a>
      <div
        className="help-message dark"
        onClick={[Function]}
      >
        <BsFillQuestionSquareFill>
          <IconBase
            attr={
              Object {
                "fill": "currentColor",
                "viewBox": "0 0 16 16",
              }
            }
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm3.496 6.033a.237.237 0 0 1-.24-.247C5.35 4.091 6.737 3.5 8.005 3.5c1.396 0 2.672.73 2.672 2.24 0 1.08-.635 1.594-1.244 2.057-.737.559-1.01.768-1.01 1.486v.105a.25.25 0 0 1-.25.25h-.81a.25.25 0 0 1-.25-.246l-.004-.217c-.038-.927.495-1.498 1.168-1.987.59-.444.965-.736.965-1.371 0-.825-.628-1.168-1.314-1.168-.803 0-1.253.478-1.342 1.134-.018.137-.128.25-.266.25h-.825zm2.325 6.443c-.584 0-1.009-.394-1.009-.927 0-.552.425-.94 1.01-.94.609 0 1.028.388 1.028.94 0 .533-.42.927-1.029.927z"
                key="0"
              />
            </svg>
          </IconBase>
        </BsFillQuestionSquareFill>
      </div>
      <div
        className="color dark"
        onClick={[Function]}
        tabIndex={1}
      />
    </div>
    <div
      className="body"
    >
      <div
        className="search"
      >
        <ReactFinalForm
          onSubmit={[Function]}
        >
          <form
            autoComplete="off"
            onSubmit={[Function]}
          >
            <h2>
              Word Finder for Wordle
            </h2>
            <h4>
              Letters in the correct position
            </h4>
            <ForwardRef(Field)
              component="input"
              key="0"
              maxLength="1"
              name="letter1"
              onFocus={[Function]}
              onInput={[Function]}
              onKeyUp={[Function]}
              parse={[Function]}
            >
              <input
                maxLength="1"
                name="letter1"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyUp={[Function]}
                value=""
              />
            </ForwardRef(Field)>
            <ForwardRef(Field)
              component="input"
              key="1"
              maxLength="1"
              name="letter2"
              onFocus={[Function]}
              onInput={[Function]}
              onKeyUp={[Function]}
              parse={[Function]}
            >
              <input
                maxLength="1"
                name="letter2"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyUp={[Function]}
                value=""
              />
            </ForwardRef(Field)>
            <ForwardRef(Field)
              component="input"
              key="2"
              maxLength="1"
              name="letter3"
              onFocus={[Function]}
              onInput={[Function]}
              onKeyUp={[Function]}
              parse={[Function]}
            >
              <input
                maxLength="1"
                name="letter3"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyUp={[Function]}
                value=""
              />
            </ForwardRef(Field)>
            <ForwardRef(Field)
              component="input"
              key="3"
              maxLength="1"
              name="letter4"
              onFocus={[Function]}
              onInput={[Function]}
              onKeyUp={[Function]}
              parse={[Function]}
            >
              <input
                maxLength="1"
                name="letter4"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyUp={[Function]}
                value=""
              />
            </ForwardRef(Field)>
            <ForwardRef(Field)
              component="input"
              key="4"
              maxLength="1"
              name="letter5"
              onFocus={[Function]}
              onInput={[Function]}
              onKeyUp={[Function]}
              parse={[Function]}
            >
              <input
                maxLength="1"
                name="letter5"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyUp={[Function]}
                value=""
              />
            </ForwardRef(Field)>
            <h4>
              Letters in any position
            </h4>
            <ForwardRef(Field)
              component="input"
              key="0"
              maxLength="1"
              name="anyPosLetter1"
              onFocus={[Function]}
              onInput={[Function]}
              onKeyUp={[Function]}
              parse={[Function]}
            >
              <input
                maxLength="1"
                name="anyPosLetter1"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyUp={[Function]}
                value=""
              />
            </ForwardRef(Field)>
            <ForwardRef(Field)
              component="input"
              key="1"
              maxLength="1"
              name="anyPosLetter2"
              onFocus={[Function]}
              onInput={[Function]}
              onKeyUp={[Function]}
              parse={[Function]}
            >
              <input
                maxLength="1"
                name="anyPosLetter2"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyUp={[Function]}
                value=""
              />
            </ForwardRef(Field)>
            <ForwardRef(Field)
              component="input"
              key="2"
              maxLength="1"
              name="anyPosLetter3"
              onFocus={[Function]}
              onInput={[Function]}
              onKeyUp={[Function]}
              parse={[Function]}
            >
              <input
                maxLength="1"
                name="anyPosLetter3"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyUp={[Function]}
                value=""
              />
            </ForwardRef(Field)>
            <ForwardRef(Field)
              component="input"
              key="3"
              maxLength="1"
              name="anyPosLetter4"
              onFocus={[Function]}
              onInput={[Function]}
              onKeyUp={[Function]}
              parse={[Function]}
            >
              <input
                maxLength="1"
                name="anyPosLetter4"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyUp={[Function]}
                value=""
              />
            </ForwardRef(Field)>
            <ForwardRef(Field)
              component="input"
              key="4"
              maxLength="1"
              name="anyPosLetter5"
              onFocus={[Function]}
              onInput={[Function]}
              onKeyUp={[Function]}
              parse={[Function]}
            >
              <input
                maxLength="1"
                name="anyPosLetter5"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyUp={[Function]}
                value=""
              />
            </ForwardRef(Field)>
            <h4>
              Letters not in answer
            </h4>
            <ForwardRef(Field)
              component="input"
              key="0"
              maxLength="1"
              name="notLetter1"
              onFocus={[Function]}
              onInput={[Function]}
              onKeyUp={[Function]}
              parse={[Function]}
            >
              <input
                maxLength="1"
                name="notLetter1"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyUp={[Function]}
                value=""
              />
            </ForwardRef(Field)>
            <br />
            <button
              className="search-button"
              type="submit"
            >
              Search
            </button>
            <br />
            <button
              className="reset-button"
              onClick={[Function]}
              type="button"
            >
              Reset
            </button>
          </form>
        </ReactFinalForm>
      </div>
      <div
        className="results"
      />
    </div>
  </div>
</Finder>
`;
